src/auth/SignupButtons.tsx

```diff
@@ -13,24 +16,56 @@ const userCreateMutation = gql`
  }} 
 `
 
+const userLoginMutation = gql`
+ mutation UserLoginMutation($username: String!) {
+ userLogin(username: $username) {
+   id
+   username
+ }} 
+`
+
 export const CreateUserButton: FC<AuthProps> = ({ username }) => {
-  const [userCreate, {data, loading}] = useMutation(userCreateMutation);
+  const [userCreate, { data, loading }] = useSWMutation<UserCreateMutation, UserCreateMutationVariables>(userCreateMutation);
+  const [_, setData] = useLocalData();
+
+  useEffect(() => {
+    if (!data) return;
+
+    setData({ name: data.userCreate?.username, id: data.userCreate?.id })
 
-  if(loading) {
+  }, [data])
+
+  if (loading) {
     return <CircularProgress />
   }
   return (
-    <Button variant="contained" color="primary" onClick={()=>userCreate({variables: {username}})}>
-      Create new user
-    </Button>
+    <>
+      <Button variant="contained" color="primary" onClick={() => userCreate({ variables: { username } })}>
+        Create new user
+      </Button>
+    </>
   );
 };
 
 export const LoginAsUser: FC<AuthProps> = ({ username }) => {
+  const [userLogin, { data, loading }] = useSWMutation<UserLoginMutation, UserLoginMutationVariables>(userLoginMutation);
+  const [_, setData] = useLocalData();
```

Comment:
Hvorfor ikke benytte en quary i stedet for en mutation, når formålet er at lave en quary i databasen?